internal/client.go

```diff
@@ -835,6 +839,33 @@ type (
 		NonRetryableErrorTypes []string
 	}
 
+	// Priority contains metadata that controls the relative ordering of task processing
+	// when tasks are backed up in a queue. The affected queues depend on the
+	// server version.
+	//
+	// Priority is attached to workflows and activities. By default, activities
+	// inherit Priority from the workflow that created them, but may override fields
+	// when an activity is started or modified.
+	//
+	// For all fields, the field not present or equal to zero/empty string means to
+	// inherit the value from the calling workflow, or if there is no calling
+	// workflow, then use the default value.
+	//
+	// Exposed as: [go.temporal.io/sdk/temporal.Priority]
+	Priority struct {
+		// PriorityKey is a positive integer from 1 to n, where smaller integers
+		// correspond to higher priorities (tasks run sooner). In general, tasks in
+		// a queue should be processed in close to priority order, although small
+		// deviations are possible.
```

Comment:
Probably worth mentioning what will happen if it's set <1. My vote would be to just clamp it to 1.

Reply:
`0` means "use default". I'm not sure what negative would mean, we could say also default

Reply:
What do you think about returning "invalid argument" instead? Removes the ambiguity. Users can't know what `n` is since it's based on the server config, but they do know that <1 is invalid.

Reply:
Are you suggesting returning invalid argument only for < 0, and accepting > n? That's fine.

I don't want to reject > n (where n is the current max) because: 1. n belongs to matching service and frontend doesn't necessarily know about it, 2. n can change over time

Reply:
@dnr yup, that's exactly what I thought 👍

Reply:
I'm not sure we need to call out the "negative number" case here explicitly, though, as the docs already state it's either default (zero) or a positive number - implying that a negative is not allowed.

Looking at other doc strings, such as `MaximumAttempts` or `BackoffCoefficient`, they aren't spelling that out either.

Reply:
I agree it's not worth mentioning in comments, "positive integer" is enough. We'll do basic validation on these in frontend and can reject negative there.

Reply:
Rejecting in the frontend sounds great